xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -725,6 +744,72 @@ ProcColorSpaces CProcessorHD::CalculateDXGIColorSpaces(const DXGIColorSpaceArgs&
   const bool limited = DX::Windowing()->UseLimitedColor();
   const bool supportHDR = DX::Windowing()->IsHDROutput() && (m_bSupportHDR10Limited || !limited);
 
-  return ProcColorSpaces{GetDXGIColorSpaceSource(csArgs, supportHDR, m_bSupportHLG),
-                         GetDXGIColorSpaceTarget(csArgs, supportHDR, limited)};
+  return ProcColorSpaces{
+      GetDXGIColorSpaceSource(csArgs, supportHDR, m_bSupportHLG, m_BT2020TopLeft),
+      GetDXGIColorSpaceTarget(csArgs, supportHDR, limited)};
+}
+
+bool CProcessorHD::IsBT2020Supported()
+{
+  ComPtr<ID3D11VideoDevice> pVideoDevice;
+  ComPtr<ID3D11VideoProcessorEnumerator> pEnumerator;
+  ComPtr<ID3D11VideoProcessorEnumerator1> pEnumerator1;
+  ComPtr<ID3D11Device> pD3DDevice = DX::DeviceResources::Get()->GetD3DDevice();
+
+  if (FAILED(pD3DDevice.As(&pVideoDevice)))
+  {
+    CLog::LogF(LOGDEBUG, "failed to get video device.");
+    return false;
+  }
+
+  D3D11_VIDEO_PROCESSOR_CONTENT_DESC desc = {};
+  desc.InputWidth = 3840;
+  desc.InputHeight = 2160;
+  desc.OutputWidth = 3840;
+  desc.OutputHeight = 2160;
+
```

Comment:
I don't know graphic drivers internals and whether picture size actually has an effect on enumerator creation, but hardcoding doesn't seem right.
4k may be too much to ask of an older gen gpu and may be rejected even if you only want to play 480p for example, but hardcoding a lower res. may also be problematic, a driver may support 1080p but not 4k.

Reply:
I don't think exist a 480p / SD video with BT.2020 color space and we want discard old HW to fallback Pixel Shaders early.

Reply:
Sure 480p is extreme for files. 1080p HEVC HDR is common, that's what I tested before.

Reply:
Tested with random numbers as 20,000 and not matters, works same, then is not really used here. Probably this only makes sense when video processor created to reserve/allocate memory, but is not the case here because is not created, only is obtained the interface `ID3D11VideoProcessorEnumerator` to obtain `ID3D11VideoProcessorEnumerator1` that implements `CheckVideoProcessorFormatConversion`.

Reply:
Yeah difficult to know for sure without knowledge of the internals of D3D11 and drivers. That's why it doesn't feel right but maybe there is no impact at all.